handlers/api.go

```diff
@@ -1649,16 +1722,21 @@ func ApiValidatorIncomeDetailsHistory(w http.ResponseWriter, r *http.Request) {
 		Epoch          uint64                       `json:"epoch"`
 		ValidatorIndex uint64                       `json:"validatorindex"`
 		Week           uint64                       `json:"week"`
+		WeekStart      time.Time                    `json:"week_start"`
+		WeekEnd        time.Time                    `json:"week_end"`
 	}
 	responseData := make([]*responseType, 0, len(history)*101)
 
+	epochsPerWeek := utils.EpochsPerDay() * 7
 	for validatorIndex, epochs := range history {
 		for epoch, income := range epochs {
 			responseData = append(responseData, &responseType{
+				Income:         income,
 				Epoch:          epoch,
 				ValidatorIndex: validatorIndex,
-				Week:           epoch / 1575,
-				Income:         income,
+				Week:           epoch / epochsPerWeek,
+				WeekStart:      utils.EpochToTime((epoch / epochsPerWeek) * epochsPerWeek),
+				WeekEnd:        utils.EpochToTime((epoch/epochsPerWeek)*epochsPerWeek + epochsPerWeek),
```

Comment:
**Minor:**
```suggestion
				WeekEnd:        utils.EpochToTime(((epoch / epochsPerWeek) + 1) * epochsPerWeek),
```
This is communicates better what you want to do imho.

Alternatively, you could create a new variable `epochsAtStartOfTheWeek` just below
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/a8c4a970428a6ccc3ead173ce7a7a2d8412a629e/handlers/api.go#L1730
and then do something like
```
WeekStart:      epochsAtStartOfTheWeek,
WeekEnd:        epochsAtStartOfTheWeek + epochsPerWeek,
```

Applies to more of your changes.

Reply:
I preferred creating a new `epochAtStartOfTheWeek` everywhere.